worksheet_B/Mandelbrot.cpp

```diff
@@ -17,27 +29,72 @@ int main()
 	// Generate the image
 	for (int pixelY = 0; pixelY < image.height(); pixelY++)
 	{
-		// TODO: Map the y coordinate into the range minY to maxY
-		//double y0 =
+		
+		double y0 = ((pixelY / ImageWidth) * (maxY - minY) + minY);
 
-		for (int pixelX = 0; pixelX < image.width(); pixelX++)
+			for (int pixelX = 0; pixelX < image.width(); pixelX++)
 		{
-			// TODO: Map the x coordinate into the range minX to maxX
-			//double x0 =
+			
+				double x0 = ((pixelX / ImageWidth) * (maxX - minX) + minX);
 
-			// TODO: implement the algorithm to colour a single pixel (x0, y0) of the Mandelbrot set fractal
 			// The code below simply fills the screen with random pixels
+			double x = 0;
+			double y = 0;
+			int Iteration = 0;
+			int maxIteration = 200;
+
+			while (Iteration < maxIteration && (x*x + (y*y) < 4)) {
+				double xTemp = (x*x - y*y + x0);
+				y = 2 * x*y + y0;
+				x = xTemp;
+				Iteration = Iteration + 1;
+			}
+
+			int Red = 0;
+			int Green = 0;
+			int Blue = 0;
+			int minValue = 0;
+			int maxValue = 255;
+
+			if (Iteration <= 60)
+			{
+				Green = maxValue;
+				Red = (Iteration / 60) * maxValue;
+			}
+
+			else if (Iteration <= 120)
+			{
+				Red = maxValue;
+				Green = maxValue - (((Iteration - 60) / 60) * maxValue);
```

Comment:
I suspect you have rounding errors in these calculations as you are dividing `int`s by `int`s -- easy fix would be to use `60.0` instead of `60`